### PR TITLE
Refactor app module

### PR DESCRIPTION
**Pull request by bilbof**
This removes the `custom_container_services` and `service_mesh` variables from the `app` module.

These seem to have been required by router and grafana, but are no longer needed.

The app module no longer explicitly supports apps which can serve traffic on multiple ports. If this need arises (for example, apps which have an admin port and cannot easily be modified) I'd recommend extracting the service mesh node module call from the app module, and have the clients of the app module also create the service mesh virtual services.

### CODE REVIEW

File: terraform/modules/app/main.tf [hunk 1]
```diff
@@ -62,16 +60,17 @@ resource "aws_ecs_service" "service" {
...
   backend_virtual_service_names    = var.backend_virtual_service_names
   mesh_name                        = var.mesh_name
-  port                             = local.container_services[count.index].port
-  protocol                         = local.container_services[count.index].protocol
+  port                             = var.port
+  protocol                         = "http"
   service_discovery_namespace_id   = var.service_discovery_namespace_id
   service_discovery_namespace_name = var.service_discovery_namespace_name
-  service_name                     = local.container_services[count.index].container_service
+  service_name                     = local.subdomain
```
Comment (fredericfran-gds) [inline terraform/modules/app/main.tf:67]: I know it is effectively the same, but I would have chosen, `var.service_name` rather than `local.subdomain`

**Review by fredericfran-gds — COMMENTED** [review]
looks, good, see my one inline comment. In addition, have you tested that Grafana can still be deployed?

File: terraform/modules/app/main.tf (hunk 1 above)
Comment (bilbof) [reply terraform/modules/app/main.tf]: Good spot, I've changed this to `var.service_name`.

**Review by fredericfran-gds — APPROVED** [review]
looks good

**Pull request merged**